### ISSUE TITLE
Segfault when reloading in 0.5

### ISSUE THREAD

**Issue by Cody-G** [JuliaPy/PyCall.jl#349]
Hello, thanks for the work on this very useful package!  When calling `reload("PyCall")` (either after importing or using the module)  I'm getting a segfault.  This occurs in the tagged 0.5.0 as well as the release-0.5 branch.

**Comment by stevengj** [comment]
Yes, I can reproduce this.   Not sure if it is practical to get a forced `reload` of PyCall working due to the interaction with `libpython`, though.

**Comment by Cody-G** [comment]
I haven't had much need to reload the module, so it's not an urgent thing anyway.  I just wanted to point out the issue in case it's related to a deeper bug in Julia.  Just curious, is there a way that we can forbid reloading the module?  Maybe the reload command can just throw an error message explaining that it's not safe to reload...but I don't know of a mechanism to do this in Julia (I assume it's impossible to have a reload function specialized for a particular string).

Anyway feel free to close the issue.  Thanks!
